src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -75,11 +77,25 @@ public RepoConfiguration(RepoLocation location, String branch, List<Format> form
     }
 
     public static void setDatesToRepoConfigs(
-            List<RepoConfiguration> configs, Optional<Date> sinceDate, Optional<Date> untilDate) {
+            List<RepoConfiguration> configs, Optional<Date> cliSinceDate, Optional<Date> cliUntilDate) {
+        Date sinceDate = (cliSinceDate.isPresent()) ? cliSinceDate.get() : null;
+        Date untilDate = (cliUntilDate.isPresent()) ? cliUntilDate.get() : new Date();
+
+        // set sinceDate to one month before untilDate if sinceDate is null.
+        if (sinceDate == null) {
+            Calendar cal = Calendar.getInstance();
+            cal.setTime(untilDate);
+            cal.add(Calendar.MONTH, -1);
+            sinceDate = cal.getTime();
+            logger.info(String.format(" As \"since date\" is not specified, only commits that were made form %s to %s "
+                    + "will be captured by the program.", sinceDate.toString(), untilDate.toString()));
+        } else if (sinceDate.equals(SinceDateArgumentType.ARBITRARY_SINCE_DATE)) {
+            sinceDate = null;
```

Comment:
the logic is questionable here, where you try to set a value to sinceDate when its null and try to set it null when it contains a specific value.

Reply:
I can understand that the logic here is misleading. While I think of ways to restructure the code and the logic, these were my thoughts and approach when writing this code: -

> If the user were to specify `-s *`, the since date will ultimately be set to the first commit date. However, we have no idea when the first commit date is going to be. So my thought here was to set an arbitrary since date first, then set back to null.
> 
> Then, when the user does not specify the since date, the since date will ultimately be set to one month before until date. But this depends on the until date, and we don't know the until date until the until date has been parsed and processed. Hence I initially set sinceDate to null, and until I have processed the untilDate, then i set sinceDate to be one month before the untilDate.

Reply:
Yes logic is sound and the code works, not saying its bad, but rather it would be best if it can be more readable and less redundant code.

E.g. in the above scenario, the null should be assigned to what would've be null in the first place.
Also, in this PR, since the since date can be determined in the early stage of the program, we should avoid propagating the problem to the later part of the program.